lib/runtime/wazero/instance.go

```diff
@@ -471,18 +474,23 @@ func (i *Instance) Exec(function string, data []byte) (result []byte, err error)
 	if mem == nil {
 		panic("nil memory")
 	}
+
 	ok := mem.Write(inputPtr, data)
 	if !ok {
 		panic("write overflow")
 	}
 
+	i.Context.Allocator = allocator
+	defer func() {
+		i.Context.Allocator = nil
+	}()
```

Comment:
There is a possibility that we unlock the mutex before clearing the allocator.  I think we should probably add `i.Unlock()` to this defer and explicitly unlock the mutex on line 469 of the early return.